toloka_provider/utils/utils.py

```diff
@@ -0,0 +1,54 @@
+"""This module contains util functions."""
+import json
+import os
+import pickle
+from decimal import Decimal
+from functools import wraps
+from typing import Any, Type, TypeVar
+
+from toloka.client import structure, add_headers
+
+T = TypeVar('T')
+
+
+def serialize_if_needed(func):
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        default_xcom_backend = 'airflow.models.xcom.BaseXCom'
+        xcom_backend = os.getenv('AIRFLOW__CORE__XCOM_BACKEND', default_xcom_backend)
+        need_to_serialize = (xcom_backend == default_xcom_backend)
+        result = add_headers('airflow')(func)(*args, **kwargs)
```

Comment:
It's not quite obvious that the function does not only serializes results, but also adds additional headers to any TolokaClient request. We should either split this functionality or rename the decorator